File: CP website/Leetcode/83. Remove Duplicates from Sorted List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

// https://leetcode.com/problems/remove-duplicates-from-sorted-list/
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode t = head;
        
        while(t!=null && t.next !=null){
            if(t.val == t.next.val){
                t.next = t.next.next;
            }
            else{
                t = t.next;
            }
        }
        
        return head;
    }
}

File: CP website/Leetcode/83.remove_duplicates_from_sorted_list.py
# https://leetcode.com/problems/remove-duplicates-from-sorted-list/
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        d  = head
        
        while (d and d.next):
            
            if d.val == d.next.val:
                d.next = d.next.next
            else:
                d = d.next
                
                
        return head